TWLight/users/views.py

```diff
@@ -87,15 +87,18 @@ def get_object(self):
         get_object in order to be able to use the SelfOnly mixin.
         """
         assert "pk" in list(self.kwargs.keys())
+
         try:
-            return User.objects.get(pk=self.kwargs["pk"])
+            if self.kwargs["pk"] == None :
+                return User.objects.get(self.kwargs["pk"])
         except User.DoesNotExist:
             raise Http404
 
     def get_context_data(self, **kwargs):
         context = super(UserDetailView, self).get_context_data(**kwargs)
         context["language_form"] = SetLanguageForm(user=self.request.user)
         context["password_form"] = PasswordChangeForm(user=self.request.user)
+        context["terms_form"] = TermsForm(user_profile=self.request.user.userprofile)
```

Comment:
How does this form and addition to the user profile change the `admin` site?

Reply:
Hello Alex,
Admin site won't be changed.  Terms of use data (boolean value and date) are already included in the User profile.
![](https://i.imgur.com/yP1OdlF.png)

This code is for adding terms_form in the users/preferences template.
Please advise me if I missed something.  Thank you.